### PR TITLE
Adds a gallery example for the ASDA module

### PR DESCRIPTION
**Pull request by Deus1704**
## PR Description

This PR adds a long-needed gallery example for the ASDA module, resolving issue #45. The explanations are based on my review of the ASDA [paper](https://arxiv.org/pdf/1804.02931.pdf), and I welcome any feedback.
Here is a quick look at the final plot, showcasing the streamlines along with the identified swirls:
image

### CODE REVIEW

File: examples/detecting_swirls.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+# observational values (ia). This information can be useful for detailed analysis
+# and is calculated using the :func:`~sunkit_image.asda.get_vortex_properties` function.
+
+ve, vr, vc, ia = get_vortex_properties(vx, vy, center_edge, image=data)
+
+###########################################################################
+# Finally, we visualize the results. We will plot the Gamma1 and Gamma2 values,
+# which highlight the vortex centers and edges respectively.
+
+fig, ax = plt.subplots()
```
Comment (nabobalis) [inline examples/detecting_swirls.py:63]: Can this just be one figure?

File: examples/detecting_swirls.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+ax.set(xlabel="x", ylabel="y")
+
+fig, ax = plt.subplots()
+ax.imshow(gamma[..., 1], origin="lower")
+ax.set_title(r"$\Gamma_2$")
+ax.set(xlabel="x", ylabel="y")
+
+###########################################################################
+# Furthermore, we can create a swirl map visualization.
+# Generating a swirl map is a crucial step in analyzing the dynamics of fluid flow,
```
Comment (nabobalis) [inline examples/detecting_swirls.py:75]:
```suggestion
# Furthermore, we can create a swirl map visualization.
#
# Generating a swirl map is a crucial step in analyzing the dynamics of fluid flow,
```

File: examples/detecting_swirls.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+# Overlay swirl edges within ROI
+for edge in center_edge["edge"]:
+    edge = np.array(edge)
+    edge_in_roi = edge[(edge[:, 0] >= x_min) & (edge[:, 0] < x_max) & (edge[:, 1] >= y_min) & (edge[:, 1] < y_max)]
+    if edge_in_roi.size > 0:
+        ax.plot(edge_in_roi[:, 0] - x_min, edge_in_roi[:, 1] - y_min, "r--")
+
+ax.set_title("Magnified Swirl Map Region with Streamlines")
+ax.set(xlabel="x", ylabel="y")
+plt.show()
```
Comment (nabobalis) [inline examples/detecting_swirls.py:130]:
```suggestion
ax.set(xlabel="x", ylabel="y")
fig.tight_layout()

plt.show()
```

File: examples/detecting_swirls.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+fig, ax = plt.subplots()
+ax.imshow(gamma[..., 0], origin="lower")
+ax.set_title(r"$\Gamma_1$")
+ax.set(xlabel="x", ylabel="y")
+
+fig, ax = plt.subplots()
+ax.imshow(gamma[..., 1], origin="lower")
+ax.set_title(r"$\Gamma_2$")
+ax.set(xlabel="x", ylabel="y")
+
```
Comment (nabobalis) [inline examples/detecting_swirls.py:72]:
```suggestion
ax.set(xlabel="x", ylabel="y")

fig.tight_layout()
```

**Review by nabobalis — COMMENTED** [review]
Just a quick first pass.

File: examples/detecting_swirls.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+###########################################################################
+# This example demonstrates how to find swirls (vortices) in a 2D velocity flow field.
+# We will use precomputed flow field data from our test dataset.
+# First, we load the velocity field and additional data.
+
+vxvy = np.load(get_test_filepath("asda_vxvy.npz"))
+vx = vxvy["vx"]
+vy = vxvy["vy"]
+data = vxvy["data"]
```
[the remark this answers is not in the document]
Comment (Deus1704) [reply examples/detecting_swirls.py:26]: Sure, let me try adding a visualization and some info about the file

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks for the PR @Deus1704

**Pull request merged**